pkg/deployments/src/task.ts

```diff
@@ -41,27 +49,42 @@ export default class Task {
     return instance;
   }
 
+  async verify(name: string, address: string, constructorArguments: unknown): Promise<void> {
+    if (!this._verifier) return logger.warn('Avoiding contract verification, no verifier defined');
+    const url = await this._verifier.call(this, name, address, constructorArguments);
+    logger.success(`Verified contract ${name} at ${url}`);
+  }
+
   async instanceAt(name: string, address: string): Promise<Contract> {
     return instanceAt(this.artifact(name), address);
   }
 
-  async run(force = false): Promise<void> {
+  async run(force = false, verify = false): Promise<void> {
     const taskPath = this._fileAt(this.dir(), 'index.ts');
     const task = require(taskPath).default;
-    await task(this, force);
+    await task(this, force, verify);
   }
 
   dir(): string {
     if (!this.id) throw Error('Please provide a task deployment ID to run');
     return this._dirAt(TASKS_DIRECTORY, this.id);
   }
 
-  artifact(name: string): Artifact {
+  buildInfo(fileName: string): BuildInfo {
     const abiDir = this._dirAt(this.dir(), 'abi');
```

Comment:
This directory should likely not be called 'abi'

Reply:
let's use `build-info`, I'll rename it in a further PR